delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutes.scala

```diff
@@ -0,0 +1,143 @@
+package ch.epfl.bluebrain.nexus.delta.routes
+
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.{Directive1, _}
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.rdf.utils.JsonKeyOrdering
+import ch.epfl.bluebrain.nexus.delta.routes.marshalling.CirceUnmarshalling
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.AuthDirectives
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRejection.ProjectNotFound
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{Project, ProjectFields, ProjectRef, ProjectRejection}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.PaginationConfig
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.ProjectSearchParams
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.{searchResourceEncoder, SearchEncoder}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, Label}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.{Identities, Lens, ProjectResource, Projects}
+import kamon.instrumentation.akka.http.TracingDirectives.operationName
+import monix.bio.IO
+import monix.execution.Scheduler
+
+/**
+  * The project routes
+  * @param identities the identity module
+  * @param projects   the projects module
+  */
+final class ProjectsRoutes private (identities: Identities, projects: Projects)(implicit
+    baseUri: BaseUri,
+    paginationConfig: PaginationConfig,
+    s: Scheduler,
+    cr: RemoteContextResolution,
+    ordering: JsonKeyOrdering
+) extends AuthDirectives(identities)
+    with DeltaDirectives
+    with CirceUnmarshalling {
+
+  import baseUri._
+
+  private val projectsIri = endpoint.toIri / "projects"
+
+  implicit val iriResolver: Lens[ProjectRef, Iri] = (ref: ProjectRef) =>
+    projectsIri / ref.organization.value / ref.project.value
+  implicit val projectContext: ContextValue       = Project.context
+
+  private def projectsSearchParams: Directive1[ProjectSearchParams] =
+    parameter("label".as[Label].?).flatMap { organization =>
+      searchParams.tmap { case (deprecated, rev, createdBy, updatedBy) =>
+        ProjectSearchParams(organization, deprecated, rev, createdBy, updatedBy)
+      }
+    }
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractSubject { implicit subject =>
+        pathPrefix("projects") {
+          concat(
+            // List projects
+            (get & extractUri & paginated & projectsSearchParams & pathEndOrSingleSlash) { (uri, pagination, params) =>
```

Comment:
I would recommend reordering the directives such that the first two are `get` and `pathEndOrSingleSlash` to avoid computing the others if there's no match